### ISSUE TITLE
[Bug]: Transparent backgrounds rendering white on Browser Windows

### ISSUE THREAD

**Issue by VerteDinde** [electron/electron#30993]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

16.0.0-nightly.20210915

### What operating system are you using?

macOS

### Operating System Version

macOS Big Sur (also affects Windows and Linux)

### What arch are you using?

x64

### Last Known Working Electron version

16.0.0-nightly.20210903

### Expected Behavior

Setting `transparent: true` should turn a browser window transparent on creation.

### Actual Behavior

Setting `transparent: true` has a white background on creation.

### Testcase Gist URL

https://gist.github.com/106a1a501ebcdd8dc5b357b3696f4afa

### Additional Information

Does not affect 14 or 15, main only. Alt + tabbing away from and back to the window does cause the transparent background to appear, so this may be a painting/timing issue.

**Comment by theolivenbaum** [comment]
For reference, this happens for us on 14 as well - we randomly get non-transparent backgrounds on MacOS

**Comment by VerteDinde** [comment]
@olawanlejoel Can you confirm what version of 14 this started in for you - 14.0.0 or 14.0.1?

**Comment by theolivenbaum** [comment]
Wait there's a 14.0.1 already? Missed the release somehow. Let me try it and I'll let you know!

**Comment by VerteDinde** [comment]
Oh awesome, thanks! 😄   I'll take a look at 14.0.0 as well, since it seems like the problem was already happening there - but any changes you notice between 14.0.0 and 14.0.1 would be very appreciated 🙇‍♀️

**Comment by theolivenbaum** [comment]
Managed to check today, 14.0.1 fixes the issue for us!

**Comment by nathanlesage** [comment]
For me it was working in 14.0.0 and stopped working in 14.0.1. So there's always a white background on macOS despite having set the vibrancy option. And it's not just random, it's consistent. Tabbing away does not fix the background color problem.

I believe it is related to the backgroundColor fix merged for 14.0.1

**Comment by pejman1** [comment]
@VerteDinde 
Same issue in windows 10 with version 14.0.1- all versions before the latest one has no problem in windows 10

**Comment by pejman1** [comment]
https://github.com/electron/electron/pull/31003

BrowserView also has same issue . it lost transparency . using released version 15.0.0 in windows 10

**Comment by nathanlesage** [comment]
I just tried out the `16.0.0-alpha.3` and the bug is still there, so the PR did not completely fix it. Or was there any API change in between?

**Comment by pejman1** [comment]
the problem still is in version 15.1.0 with windows 10
transparency of browserview dosn't work

**Comment by 0xdeafcafe** [comment]
Present on Electron 15 on macOS

**Comment by nathanlesage** [comment]
@devinbinnie This would indicate that there might've been a change to the API, right? I can't test this on macOS right now, but it sounds like that. If so, all that would be necessary would be a documentation update. I'll see to testing this next thing tomorrow morning!

**Comment by mikeyyyzhao** [comment]
@nathanlesage when you say " You now have to additionally set the transparent flag to true on the BrowserWindow configuration", are you saying you have to set `BrowserWindow` to 

```
new BrowserWindow({
   transparent: true
});
```

 or 

```
new BrowserWindow({
   webPreferences: {
       transparent: true
  }
});
```
or something else?

I've tried the first two and still seeing the white background for my Browserwindow that has `transparent: true` after updating to 15.1.2

**Comment by VerteDinde** [comment]
Hey folks! The original issue that was logged here has been solved, so I'd like to close this specific issue. However, I see a lot of people are still having transparency issues with BrowserViews.

If it's possible, would you mind creating and attaching a repro gist that has your issue? [Electron Fiddle](https://www.electronjs.org/fiddle) is a great tool for making small test cases and makes it easy to publish your test case to a [gist](https://gist.github.com) that we maintainers can use.

I'll leave this issue open for the remainder of this week to collect the repros and see if we can more explicitly identify the problem. Once we have a better idea of what the still-open issues are, I'll close this issue and make a new one that has more explicit repro steps, so we don't confuse several different potential issues 🙂  Thanks very much!

**Comment by theolivenbaum** [comment]
If anyone is still hitting this issue: I've managed to find a way to [reproduce it](https://github.com/electron/electron/issues/31862#issuecomment-970352877): seems like Electron caches the background color of the HTML that is loaded initially by the window, so if anything causes it to be set to a non-transparent value, the window might eventually get rendered again opaque instead of transparent. 

In our case, we're hitting it due to two issues: we first preloaded our windows to "about:blank" then loaded the URL we needed (which means about:blank had a transparent background), and once loading the correct URL, there was a race-condition with styles being loaded and applied that caused the document to momentarily have a non-transparent background. Invisible to human eye but enough to randomly mess things up.
